### PR TITLE
Add prettier for doc

### PR DESCRIPTION
**Pull request by Miloas**
I found there are many unused imports in doc source code, so I tried to add prettier. 

And I think we need `pre-commit` too, we can use raw `pre-commit` or add a `project.json` at root and then use `husky`. so before we make a decision, just add prettier for now.

### CODE REVIEW

**Comment by Milo123459** [comment]
i dont think we need a precommit hook for this. just prettier is fine imo

**Comment by Miloas** [comment]
> i dont think we need a precommit hook for this. just prettier is fine imo

if we dont add rust lint and format I think that's ok, because only railway people may modify the doc source code 🤔️

File: docs/package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
     "build": "next build",
     "start": "next start",
     "lint": "next lint",
-    "clean": "rm -rf .next"
+    "clean": "rm -rf .next",
+    "format": "prettier --write \"**/*.{js,ts,tsx,md}\""
```
Comment (FinnDore) [inline docs/package.json:9]: Format:check script could be added for linting

**Comment by FinnDore** [comment]
We should add a prettier config as the rest of the repo uses 4 spaces so we should stay consistent here .

Agree with Milo here, pre commit hooks are nasty, we should check files are formatted correctly in ci instead.

File: docs/package.json (hunk 1 above)
Comment (Miloas) [reply docs/package.json]: done

**Comment by Miloas** [comment]
> 

Are you sure 4 spaces? because all the doc source code is 2 spaces based..

**Comment by Miloas** [comment]
> We should add a prettier config as the rest of the repo uses 4 spaces so we should stay consistent here .

Are you sure 4 spaces? because all the doc source code is 2 spaces based..

**Comment by FinnDore** [comment]
> > We should add a prettier config as the rest of the repo uses 4 spaces so we should stay consistent here .
> 
> Are you sure 4 spaces? because all the doc source code is 2 spaces based..

was referring to the rust side of the codebase. But if the doc's are already 2 i guess its fine? will see what milo and jr thinks

**Comment by Miloas** [comment]
cc @Milo123459 @coffee-cup

**Comment by Miloas** [comment]
BTW, if we want indentation consistent, we also can add `.editorconfig`. 
IMO, 2 spaces is fine. (actually I love 2 spaces, lol)

**Review by coffee-cup — APPROVED** [review]
Agreed that I don't think we need a pre-commit step.
Thanks for adding prettier! 🎉

**Review by coffee-cup — APPROVED** [review]
Agreed that I don't think we need a pre-commit step.
Thanks for adding prettier! 🎉

**Review by coffee-cup — APPROVED** [review]
Agreed that I don't think we need a pre-commit step.
Thanks for adding prettier! 🎉

**Pull request merged**